test/e2utils/kpm_utils.go

```diff
@@ -0,0 +1,94 @@
+// SPDX-FileCopyrightText: 2020-present Open Networking Foundation <info@opennetworking.org>
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package e2utils
+
+import (
+	"context"
+	"github.com/onosproject/onos-api/go/onos/topo"
+	"testing"
+
+	e2api "github.com/onosproject/onos-api/go/onos/e2t/e2/v1beta1"
+
+	sdkclient "github.com/onosproject/onos-ric-sdk-go/pkg/e2/v1beta1"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/onosproject/onos-e2t/test/utils"
+)
+
+const (
+	defaultReportPeriod = uint32(5000)
+	defaultGranularity  = uint32(500)
+)
+
+type KPMV2Sub struct {
```

Comment:
There is another subscription for RC PRE that should be handled. Some of these info are redundant between the two subscription and that is why we have one Generic subscription. Perhaps the Subscribe and Unsubscribe methods should be added to Subscribe structure (i.e. define an interface for that structure) and you use embedding to extend Subscribe for service model specefic subscriptions (KpmV2Sub embeds Subscribe and extends it to include more info)